src/votekit/ballot_generator.py

```diff
@@ -1201,4 +1201,144 @@ def __init__(self, **data):
         k = len(data["candidates"])
         # Call the parent class's __init__ method to handle common parameters
         super().__init__(k=k, **data)
-        
\ No newline at end of file
+        
+
+class SlatePreference(BallotGenerator):
+    """
+    Class for generating ballots using a slate preference model. 
+    This model first samples a ballot type by flipping a cohesion parameter weighted coin.
+    It then fills out the ballot type via sampling with out replacement from the interval.
+
+    Only works with 2 blocs at the moment.
+    
+    Can be initialized with an interval or can be
+    constructed with the Dirichlet distribution using the `from_params` method in the
+    `BallotGenerator` class.
+
+    **Attributes**
+
+    `slate_to_candidates`
+    :   dictionary mapping of slate to candidates (ex. {bloc: [candidate]}).
+
+    `pref_interval_by_bloc`
+    :   dictionary mapping of bloc to preference interval.
+        (ex. {bloc: {candidate : interval length}}).
+
+    `bloc_voter_prop`
+    :   dictionary mapping of bloc to voter proportions (ex. {bloc: proportion}).
+
+    `cohesion_parameters`
+    :   dictionary mapping of slate to cohesion level (ex. {bloc: .7}).
+
+    **Methods**
+
+    See `BallotGenerator` base class
+    """
+
+    def __init__(self, slate_to_candidates: dict = {}, cohesion_parameters: dict = {}, **data):
+        # Call the parent class's __init__ method to handle common parameters
+        super().__init__(**data)
+        self.slate_to_candidates = slate_to_candidates
+        self.cohesion_parameters = cohesion_parameters
+    
+    def generate_profile(self, number_of_ballots: int, by_bloc: bool = False) \
+                                                            -> Union[PreferenceProfile, Tuple]:
+        # the number of ballots per bloc is determined by Huntington-Hill apportionment
+        blocs = list(self.bloc_voter_prop.keys())
+        bloc_props = list(self.bloc_voter_prop.values())
+        ballots_per_block = dict(
+            zip(blocs, apportion.compute("huntington", bloc_props, number_of_ballots))
+        )
+
+        pref_profile_by_bloc = {}
+
+        for bloc in blocs: 
+            ballot_pool = []
+            # number of voters in this bloc
+            num_ballots = ballots_per_block[bloc]
+
+            # TODO the code below assumes only two blocs
```

Comment:
maybe add an exception if more than 2 bloc are passed in right now so code doesn't silently give incorrect results for >2 bloc